packages/core/utils/src/convert-query-params.ts

```diff
@@ -386,11 +390,33 @@ const createTransformer = ({ getModel }: TransformerOptions) => {
           );
         }
 
+        /**
+         * Validate nested population queries in the context of a polymorphic attribute (dynamic zone, morph relation).
+         *
+         * If 'populate' exists in subPopulate, its value should be constrained to a wildcard ('*').
+         *
+         * TODO: This approach should be marked as deprecated and replaced with a more robust solution in future versions.
+         *       e.g. Using glob patterns in the fragment API.
+         */
+        if ('populate' in subPopulate && subPopulate.populate !== '*') {
+          throw new Error(
+            `Invalid nested population query detected. When using 'populate' within polymorphic structures, ` +
+              `its value must be '*' to indicate all second level links. Specific field targeting is not supported here. ` +
+              `Consider using the fragment API for more granular population control.`
+          );
+        }
+
         const newSubPopulate = {};
 
-        // { on: { ... } }
+        // case: { populate: '*' }
```

Comment:
Should we prevent a mix of the 3 params ? so it's only a single one ?

Reply:
Even though it doesn't make sense to have multiple params, I believe it is used by the CM for some reason.

I would advise keeping the logic as it is right now, and adjust that when we'll look at the next behavior

Reply:
Ok, can you add a // TODO to prevent this and discuss it with the content squad ?